catalyst/loggers/wandb.py

```diff
@@ -0,0 +1,180 @@
+        runner.train(
+            ...,
+            loggers={"wandb": dl.WandbLogger(project="wandb_test", name="expeirment1")}
+        )
+
+    .. code-block:: python
+
+        from catalyst import dl
+
+        class CustomRunner(dl.IRunner):
+            # ...
+
+            def get_loggers(self):
+                return {
+                    "console": dl.ConsoleLogger(),
+                    "wandb": dl.WandbLogger(project="wandb_test", config=self.hparams)
+                }
+
+            # ...
+
+        runner = CustomRunner().run()
+
+    Config API example:
+
+    .. code-block:: yaml
+
+        loggers:
+            wandb:
+                _target_: WandbLogger
+                project: test_exp
+                name: test_run
+        ...
+
+    Hydra API example:
+
+    .. code-block:: yaml
+
+        loggers:
+            wandb:
+                _target_: catalyst.dl.WandbLogger
+                project: test_exp
+                name: test_run
+        ...
+    """
+
+    def __init__(
+        self,
+        project: str,
+        name: Optional[str] = None,
+        config: Optional[Dict] = None,
+        entity: Optional[str] = None,
+    ) -> None:
+        self.project = project
+        self.name = name
+        self.config = config
+        self.entity = entity
+        self.run = wandb.init(
+            project=self.project, name=self.name, config=self.config, entity=self.entity
+        )
+
+    def _log_metrics(self, metrics: Dict[str, float], step: int, loader_key: str, suffix=""):
+        for key, value in metrics.items():
+            self.run.log({f"{key}/{loader_key}{suffix}": value}, step=step)
+
+    def log_metrics(
+        self,
+        metrics: Dict[str, Any],
+        scope: str = None,
+        # experiment info
+        run_key: str = None,
+        global_epoch_step: int = 0,
+        global_batch_step: int = 0,
+        global_sample_step: int = 0,
+        # stage info
+        stage_key: str = None,
+        stage_epoch_len: int = 0,
+        stage_epoch_step: int = 0,
+        stage_batch_step: int = 0,
+        stage_sample_step: int = 0,
+        # loader info
+        loader_key: str = None,
+        loader_batch_len: int = 0,
+        loader_sample_len: int = 0,
+        loader_batch_step: int = 0,
+        loader_sample_step: int = 0,
+    ) -> None:
+        """Logs batch and epoch metrics to wandb."""
+        if scope == "batch":
+            metrics = {k: float(v) for k, v in metrics.items()}
+            self._log_metrics(
+                metrics=metrics, step=global_batch_step, loader_key=loader_key, suffix="/batch"
+            )
+        elif scope == "epoch":
+            for loader_key, per_loader_metrics in metrics.items():
+                self._log_metrics(
+                    metrics=per_loader_metrics,
+                    step=global_epoch_step,
+                    loader_key=loader_key,
+                    suffix="/epoch",
+                )
```

Comment:
we have an update for tensorboard logger recently (we log loader metric just after the loader rather then in the end of the epoch): https://github.com/catalyst-team/catalyst/blob/master/catalyst/loggers/tensorboard.py#L122#L136
would you like to do the same for wandb one?